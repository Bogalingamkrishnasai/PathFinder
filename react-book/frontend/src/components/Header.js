import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import UserDetails from './UserDetails';
import CompanyLogo from '../images/pathfinder.png';
import CartIcon from '../images/cart.svg';
import './Header.css';

function Header({ cart, dispatch }) {
  const cartQuantity = cart.reduce((acc, item) => acc + item.quantity, 0);

  return (
    <header className="header-component">
      <Link to="/" className="brand-link">
        <img src={CompanyLogo} alt="Pathfinder Logo" className="logo" />
        <h1 className="brand-name">Pathfinder</h1>
      </Link>
      <div className="search-bar">
        <input type="text" placeholder="Search..." className="search-input" />
        <button type="button" className="search-button">Search</button>
      </div>
      <div className="menu">
        <Link to="/cart" className="cart-link">
          My Cart
          <img src={CartIcon} alt="Cart" className="cart-icon" />
          <div className="cart-badge">{cartQuantity}</div>
        </Link>
        <UserDetails dispatch={dispatch} />
      </div>
    </header>
  );
}

Header.propTypes = {
  cart: PropTypes.arrayOf(
    PropTypes.shape({
      itemId: PropTypes.string.isRequired,
      quantity: PropTypes.number.isRequired,
    }),
  ).isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default Header;
