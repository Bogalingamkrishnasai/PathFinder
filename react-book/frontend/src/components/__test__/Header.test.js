import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from '../Header';
import '@testing-library/jest-dom';
import React from 'react';

// 1. Create a mock context directly in the test file
const CurrentUserContext = React.createContext({
  currentUser: null,
  setCurrentUser: jest.fn()
});

// 2. Mock the UserDetails component to avoid context issues
jest.mock('../UserDetails', () => () => <div>Mock UserDetails</div>);

describe('Header Component', () => {
  const mockCart = [
    { itemId: '1', quantity: 2 },
    { itemId: '2', quantity: 1 }
  ];

  // Helper function to render the component
  const renderHeader = () => {
    return render(
      <CurrentUserContext.Provider value={{ currentUser: null, setCurrentUser: jest.fn() }}>
        <BrowserRouter>
          <Header cart={mockCart} />
        </BrowserRouter>
      </CurrentUserContext.Provider>
    );
  };

  test('renders header with logo and brand name', () => {
    renderHeader();
    
    expect(screen.getByAltText('Pathfinder Logo')).toBeInTheDocument();
    expect(screen.getByText('Pathfinder')).toBeInTheDocument();
  });

  test('renders search bar', () => {
    renderHeader();
    
    expect(screen.getByPlaceholderText('Search...')).toBeInTheDocument();
    expect(screen.getByText('Search')).toBeInTheDocument();
  });

  test('displays correct cart quantity', () => {
    renderHeader();
    
    expect(screen.getByText('3')).toBeInTheDocument();
  });

  test('has navigation links', () => {
    renderHeader();
    
    expect(screen.getByText('My Cart')).toBeInTheDocument();
    expect(screen.getByText('My Cart')).toHaveAttribute('href', '/cart');
  });

  test('matches snapshot', () => {
    const { asFragment } = renderHeader();
    expect(asFragment()).toMatchSnapshot();
  });
});