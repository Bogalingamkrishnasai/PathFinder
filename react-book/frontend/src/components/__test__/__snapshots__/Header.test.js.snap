// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header-component"
  >
    <a
      class="brand-link"
      href="/"
    >
      <img
        alt="Pathfinder Logo"
        class="logo"
        src="test-file-stub"
      />
      <h1
        class="brand-name"
      >
        Pathfinder
      </h1>
    </a>
    <div
      class="search-bar"
    >
      <input
        class="search-input"
        placeholder="Search..."
        type="text"
      />
      <button
        class="search-button"
        type="button"
      >
        Search
      </button>
    </div>
    <div
      class="menu"
    >
      <a
        class="cart-link"
        href="/cart"
      >
        My Cart
        <img
          alt="Cart"
          class="cart-icon"
          src="test-file-stub"
        />
        <div
          class="cart-badge"
        >
          3
        </div>
      </a>
      <div>
        Mock UserDetails
      </div>
    </div>
  </header>
</DocumentFragment>
`;
